static/app/views/settings/components/teamSelect.tsx

```diff
@@ -55,53 +57,114 @@ type Props = {
    * Optional menu header.
    */
   menuHeader?: React.ReactElement;
+  /**
+   * Used when showing Teams for a Member
+   */
+  onChangeTeamRole?: (teamSlug: string, teamRole: string) => void;
+  /**
+   * Used when showing Teams for a Member
+   */
+  selectedOrgRole?: Member['orgRole'];
+  /**
+   * Used when showing Teams for a Member
+   */
+  selectedTeamRoles?: Member['teamRoles'];
+  /**
+   * Used when showing Teams for a Project
+   */
+  selectedTeams?: Team[];
 };
 
 function TeamSelect({
   disabled,
+  loadingTeams,
   enforceIdpProvisioned,
-  selectedTeams,
   menuHeader,
+  confirmLastTeamRemoveMessage,
+  selectedOrgRole,
+  selectedTeamRoles,
+  selectedTeams,
   organization,
   onAddTeam,
   onRemoveTeam,
-  confirmLastTeamRemoveMessage,
-  loadingTeams,
+  onChangeTeamRole,
 }: Props) {
   const {teams, onSearch, fetching} = useTeams();
-
-  const handleAddTeam = (option: Item) => {
-    const team = teams.find(tm => tm.slug === option.value);
-    if (team) {
-      onAddTeam(team);
-    }
-  };
+  const {orgRoleList, teamRoleList} = organization;
 
   const renderBody = () => {
```

Comment:
nit: can we move all `render` methods down at the end of the class just organize things a bit?